app/scripts/inpage.js

```diff
@@ -34,7 +34,7 @@ cleanContextForImports();
 import log from 'loglevel';
 import { v4 as uuid } from 'uuid';
 import { WindowPostMessageStream } from '@metamask/post-message-stream';
-import { initializeProvider } from '@metamask/providers/dist/initializeInpageProvider';
+import { initializeProvider } from '../../node_modules/@metamask/providers/dist/initializeInpageProvider.cjs';
```

Comment:
The reason why normal importing didn't work might have something to do with the still incomplete exports support in some of the tools we are using. 
https://github.com/browserify/resolve/pull/224

Reply:
> have you tried `@metamask/providers/dist/initializeInpageProvider.cjs` ?

Yes. For some reason that works for `@metamask/providers` [v17](https://github.com/MetaMask/metamask-extension/pull/27049) but not for v18 (here).